lib/srv/discovery/fetchers/aws-sync/cloudwatchlogs.go

```diff
@@ -0,0 +1,108 @@
+/*
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package aws_sync
+
+import (
+	"context"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatchlogs"
+	cwltypes "github.com/aws/aws-sdk-go-v2/service/cloudwatchlogs/types"
+	"github.com/gravitational/trace"
+
+	accessgraphv1alpha "github.com/gravitational/teleport/gen/proto/go/accessgraph/v1alpha"
+)
+
+// cwlClient has the AWS interfaces required by this package to fetch
+// CloudWatch logs.
+type cwlClient interface {
+	cloudwatchlogs.FilterLogEventsAPIClient
+}
+
+// FetchEKSAuditLogs returns a slice of audit log events for the given cluster
+// starting from the given cursor.
+func (a *Fetcher) FetchEKSAuditLogs(ctx context.Context, cluster *accessgraphv1alpha.AWSEKSClusterV1, cursor *accessgraphv1alpha.KubeAuditLogCursor) ([]cwltypes.FilteredLogEvent, error) {
+	cfg, err := a.AWSConfigProvider.GetConfig(ctx, cluster.Region, a.getAWSOptions()...)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	client := a.awsClients.getCloudWatchLogsClient(cfg)
+
+	// limit is not a hard limit - we may exceed it but won't get any more pages
+	// once reached.
+	var limit int32 = 500 // TODO(camscale): Consider making this a parameter
+	startTime := cursor.LastEventTime.AsTime().UTC()
+	input := &cloudwatchlogs.FilterLogEventsInput{
+		LogGroupName:        aws.String("/aws/eks/" + cluster.Name + "/cluster"),
+		LogStreamNamePrefix: aws.String("kube-apiserver-audit-"),
+		StartTime:           aws.Int64(startTime.UnixMilli()),
+		Limit:               aws.Int32(limit),
+	}
+
+	var result []cwltypes.FilteredLogEvent
+	for p := cloudwatchlogs.NewFilterLogEventsPaginator(client, input); p.HasMorePages(); {
+		output, err := p.NextPage(ctx)
```

Comment:
check for ctx.Err() above? AI suggests it, I could not trace to wether this is really an issue